ckanext/validation/logic.py

```diff
@@ -661,6 +661,13 @@ def _run_sync_validation(resource_id, local_upload=False, new_resource=True):
 
     if validation['report']:
         report = json.loads(validation['report'])
+    # Check upload of invalid data is allowed
+    allow_invalid_data = bool(t.config.get(
+        'ckanext.validation.allow_invalid_data'
+    ))
+
+    if not report['valid'] and not allow_invalid_data:
```

Comment:
Why you cannot make the `allow_invalid_data` check in line 672 inside the `if validation['report']` block?
In my head extending the line :672 to:
```
if not report['valid'] and not allow_invalid_data:
```
should be all that we need.